File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ret_dict = defaultdict(int)

        for i in nums:
            if (i in ret_dict):
                ret_dict[i] += 1
            else:
                ret_dict[i] = 1
        
        return sorted(ret_dict, key=ret_dict.get, reverse=True)[:k]
        

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cs
public class Solution {
    public int FindDuplicate(int[] nums) {
        int sp1 = 0;
        int fp = 0;
        int sp2 = 0;

        sp1 = nums[0];
        fp = nums[nums[0]];
        while (sp1 != fp) {
            sp1 = nums[sp1];
            fp = nums[nums[fp]];

            if (sp1 == fp) {
                break;
            }
        }

        while (sp1 != sp2) {
            sp1 = nums[sp1];
            sp2 = nums[sp2];

            if (sp1 == sp2) {
                break;
            }
        }

        return sp1;
    }
}
